packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1564,6 +1564,13 @@ export default class TradeStore extends BaseStore {
     onMount() {
         this.root_store.notifications.removeTradeNotifications();
         if (this.is_trade_component_mounted && this.should_skip_prepost_lifecycle) {
+            const { chart_type, granularity } = this.root_store.contract_trade;
+            setTradeURLParams({
+                chartType: chart_type,
+                granularity,
+                symbol: this.symbol,
+                contractType: this.contract_type,
+            });
```

Comment:
This is fixing the issue in the following case: `DTrader page -> C.Details page -> Close button click -> DTrader page`, - when we end up inside this condition.
At this point, `Trade` component (DTrader page) is already mounted and all the state is loaded and available. Hence, we can set the DTrader URL params to the existing values before we stop are `return` below.